entc/gen/template/client.tmpl

```diff
@@ -106,6 +111,42 @@ func (c *Client) Debug() *Client {
 	}
 }
 
+// DebugWithContext returns a new debug-client. It's used to get verbose 
+// logging with context on specific operations.
+// To enable contextual logging, logger which can log with context must be configured.
+// 
+//	// using github.com/rs/zerolog
+//	logCtx := func(ctx context.Context, i ...interface{}) {
+//		log.Ctx(ctx).Print(i)
+//	}
+//
+//	client, _ := ent.Open("sqlite3", "file:ent?mode=memory&cache=shared&_fk=1", ent.LogCtx(logCtx))
+//
+//	client.DebugWithContext().
+//		{{ (index $.Nodes 0).Name }}.
+//		Query().
+//		Count(ctx)
+//
+func (c *Client) DebugWithContext() *Client {
```

Comment:
I'm not sure I like it. 
Since, this is not so common, we can keep the API as it is today, and let people using it as follows: 

```go
func main() {
    drv, err := sql.Open("<dialect>", "<mysql-dsn>")
    if err != nil {
        return nil, err
    }
    drv := dialect.DebugWithContext(drv, logf)
    client := ent.NewClient(ent.Driver(drv))
}
```

Reply:
That would be enough. Removed config.